### ISSUE TITLE
Can't get viz.view() to work

### ISSUE THREAD

**Issue by jlw365** [parrt/dtreeviz#23]
Potentially related to #19 

I installed dot using your command `brew install graphviz --with-librsvg --with-app --with-pango` and also have done `pip install graphviz`. I am having trouble with the `viz.view()` command when trying to run your example code. I am getting the following error:

```Format: "svg:cairo" not recognized. Use one of: svg:svg:core
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
<ipython-input-21-4930bdb0a4e7> in <module>()
     12                X=X)  # need to give single observation for prediction
     13 
---> 14 viz.view()

/anaconda3/lib/python3.6/site-packages/dtreeviz/trees.py in view(self)
     66         tmp = tempfile.gettempdir()
     67         svgfilename = f"{tmp}/DTreeViz_{getpid()}.svg"
---> 68         self.save(svgfilename)
     69         view(svgfilename)
     70 

/anaconda3/lib/python3.6/site-packages/dtreeviz/trees.py in save(self, filename)
     89             cmd = ["dot", f"-T{format}:cairo", "-o", filename, dotfilename]
     90             # print(' '.join(cmd))
---> 91             stdout, stderr = run(cmd, capture_output=True, check=True, quiet=False)
     92 
     93         else:

/anaconda3/lib/python3.6/site-packages/graphviz/backend.py in run(cmd, input, capture_output, check, quiet, **kwargs)
    157         stderr_write_bytes(err, flush=True)
    158     if check and proc.returncode:
--> 159         raise CalledProcessError(proc.returncode, cmd, output=out, stderr=err)
    160 
    161     return out, err

CalledProcessError: Command '['dot', '-Tsvg:cairo', '-o', '/var/folders/13/x7py_p6115gdj9y7_bybd8xc4gmyvs/T/DTreeViz_84409.svg', '/var/folders/13/x7py_p6115gdj9y7_bybd8xc4gmyvs/T/DTreeViz_84409']' returned non-zero exit status 1.
```

I am running Python 3.6.5::Anaconda in a Jupyter Notebook. I also tried this in Jupyter Lab and got the same error. Thanks for your help!

**Comment by parrt** [comment]
weird. does `dot` work from command line?

**Comment by p9anand** [comment]
getting same issue.  Although i have the dot.
<img width="335" alt="screen shot 2018-12-06 at 3 52 45 pm" src="https://user-images.githubusercontent.com/7848315/49578181-25765c00-f96f-11e8-9223-7018c0f50455.png">

**Comment by parrt** [comment]
Try running the code from cmdline then not notebook to see if it works. that will isolate the issue to a path problem in notebook.

**Comment by p9anand** [comment]
Hi Thanks for reverting back. still having same issue. Please find attached file.
<img width="1440" alt="screenshot 2018-12-07 at 10 50 05 pm" src="https://user-images.githubusercontent.com/7848315/49662698-92bee580-fa72-11e8-8c9e-6418ca89d9a2.png">

**Comment by parrt** [comment]
Ah. ok, it's finding dot but can't handle the svg stuff. That is a mac?  try these updated instructions:

Make sure to have the latest XCode installed and command-line tools installed. You can run xcode-select --install from the command-line to install those if XCode is already installed. You also have to sign the XCode license agreement, which you can do with sudo xcodebuild -license from command-line. The brew install shown next needs to build graphviz, so you need XCode set up properly.

You need the graphviz binary for dot installed with librsvg and pango. Make sure you uninstall graphviz then reinstall or install like this:

```
brew install graphviz --with-librsvg --with-pango
```

(The --with-librsvg is absolutely required because we generate output using dot's -Tsvg:cairo option. I have used --with-app successfully also but others have trouble installing with that.)

**Comment by motoki0214** [comment]
I struggled the same issue then I could solve them with the information here.
Thank you so much for your helpful information, @parrt.

**Comment by parrt** [comment]
@motoki0214 glad to hear it. What specifically solved your problem so I can update the doc? Was the uninstall then install?  correct brew options?

**Comment by motoki0214** [comment]
@parrt 
I did what you said things below,
1, type "xcode-select --install" from the command-line
2, then run "sudo xcodebuild -license" from the command-line
3, "uninstall graphviz" from the command-line once
4, then finally run "brew install graphviz --with-librsvg --with-pango"
5, A sample decision tree was shown on Jupyter notebook with "display(viz)"

FYI ;
I am running Python 3.6.4 :: Anaconda, Inc. in the Jupyter notebook.
also already Xcode was installed in my mac.

**Comment by parrt** [comment]
great. added to doc

**Comment by motoki0214** [comment]
@parrt 
My pleasure.
This library is very useful for my job, thank you very much too.

**Comment by JosephGatto** [comment]
I have tried all of the suggestions from all of these threads, everything installs successfully yet I still get:
CalledProcessError: Command '['dot', '-Tsvg:cairo', '-o',  .... ]   (on mac) 

Since I am not too worried about quality, my work around was to go to the trees.py file and change line 89 to:
cmd = ["dot", f"**-Tsvg:svg:core**", "-o", filename, dotfilename]

<img width="664" alt="git" src="https://user-images.githubusercontent.com/35414926/51066563-d3ed6c80-15bf-11e9-9260-5d111e49fdcd.png">

Now this doesn't fix viz.view() for me but it does allow me to save .png. Better than nothing! 

--
By the way, awesome library @parrt thank you

**Comment by parrt** [comment]
That is so weird. If you try `dot`  with svg from the command line and it doesn't work, then that is some kinda weird install issue unrelated to dtreeviz.  Nonetheless, glad you found a way around it.

**Comment by parrt** [comment]
@JosephGatto That -Tsvg:svg:core thing didn't work for me.  How did you use that route to get png? Trying to update package. thanks!

**Comment by JosephGatto** [comment]
> @JosephGatto That -Tsvg:svg:core thing didn't work for me. How did you use that route to get png? Trying to update package. thanks! This seems to work though:
> 
> ```
> dot -o foo.png -Tpng foo.dot
> ```

@parrt So my reasoning for that was Jupyter warnings stating something telling me to use that package instead. I then just went into the source code and replaced the original call. 

It worked for a bit but I have since tried uninstalling / reinstalling dtreeviz as that workflow was painful. My hacky solution stopped working after that and I resulted in giving up on trying to get it to work on my mac and switched to windows. Sorry this response isn't more helpful!
